### PR TITLE
Iosxe/show cts role based permissions

### PR DESCRIPTION
**Pull request by reustaquio22**

### CODE REVIEW

File: changelog/undistributed.rst [hunk 1]
```diff
@@ -14,8 +14,10 @@
         * show stackwise-virtual
     * Added ShowCtsPacs
         * show cts pacs
-    * Added ShowClassMap
-        * show class-map
+    * Added ShowCtsRoleBasedPermissions
+        * show cts role-based permissions
+    * Added ShowCtsRoleBasedPermissions
+        * show cts role-based permissions
```
Comment (tahigash) [inline changelog/undistributed.rst]: Why twice?
```suggestion
    * Added ShowCtsRoleBasedPermissions
        * show cts role-based permissions
```

File: src/genie/libs/parser/iosxe/tests/test_show_cts.py [hunk 2]
```diff
@@ -51,6 +52,153 @@ def test_show_cts_pacs_empty(self):
...
+# Unit test for 'show cts role-based permissions'
+# =============================
+class TestShowCtsRoleBasedPermissions(unittest.TestCase):
+    """Unit test for 'show cts role-based permissions'"""
+
+    maxDiff = None
+    empty_output = {'execute.return_value': ''}
+    golden_parsed_output1 = {
+        "rbp_policies": {
+            1: {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/test_show_cts.py:65]: I believe this number is index. if so, better to have key name as `indexes` like below:
```
"indexes": {
    1: {
        "policy_name": "default",
        (snip)

File: src/genie/libs/parser/iosxe/tests/test_show_cts.py [hunk 3]
```diff
@@ -51,6 +52,153 @@ def test_show_cts_pacs_empty(self):
...
+                "dst_group_name": "Untrusted",
+                "action_policy": "Deny",
+                "action_policy_group": "IP-00"
+            },
+            5: {
+                "src_grp_id": 100,
+                "src_grp_name": "Tech",
+                "dst_group_id": 6,
+                "dst_group_name": "Trusted",
+                "policy_groups": {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/test_show_cts.py:97]: no issue to have list at leaf of schema. so, can change to below to be simpler?
```
"policy_groups": ['NFS-08', 'ACCESS-01']

**Review by tahigash — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/tests/test_show_cts.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                "action_policy_group": "IP-00"
+            },
+            "monitor_dynamic": False,
+            "monitor_configured": False
+        }
+    }
+
+    golden_output1 = {'execute.return_value': '''
+IPv4 Role-based permissions default:
+        Permit IP-00
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/test_show_cts.py:163]: do you know if any chance to have multiple like below? if yes, we might need to consider to update schema for it.
```
IPv4 Role-based permissions default:
        Permit IP-00
        Permit IP-01

**Comment by itdependsnetworks** [comment]
please close, covered in https://github.com/CiscoTestAutomation/genieparser/pull/250

File: changelog/undistributed.rst (hunk 1 above)
Comment (tahigash) [reply changelog/undistributed.rst]: confirmed in https://github.com/CiscoTestAutomation/genieparser/pull/250

File: src/genie/libs/parser/iosxe/tests/test_show_cts.py (hunk 2 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_cts.py]: added the same to https://github.com/CiscoTestAutomation/genieparser/pull/250

File: src/genie/libs/parser/iosxe/tests/test_show_cts.py (hunk 4 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_cts.py]: please ignore. added the same to https://github.com/CiscoTestAutomation/genieparser/pull/250

File: src/genie/libs/parser/iosxe/show_cts.py [hunk 5]
```diff
@@ -149,3 +149,166 @@ def filter_lines(raw_output, remove_lines):
...
+        # ACCESS-01
+        policy_group_capture = re.compile(r"^(?P<policy_group>\w+-\d+)")
+        # RBACL Monitor All for Dynamic Policies : FALSE
+        monitor_dynamic_capture = re.compile(
+            r"^RBACL\s+Monitor\s+All\s+for\s+Dynamic\s+Policies\s+:\s+(?P<monitor_dynamic>(TRUE|FALSE))")
+        #RBACL Monitor All for Configured Policies : FALSE
+        monitor_configured_capture = re.compile(
+            r"^RBACL\s+Monitor\s+All\s+for\s+Configured\s+Policies\s+:\s+(?P<monitor_configured>(TRUE|FALSE))")
+
+        remove_lines = ()
```
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_cts.py]: added the same to https://github.com/CiscoTestAutomation/genieparser/pull/250

File: src/genie/libs/parser/iosxe/tests/test_show_cts.py (hunk 3 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_cts.py]: added the same to https://github.com/CiscoTestAutomation/genieparser/pull/250

**Comment by tahigash** [comment]
as per request, closed.

**Pull request closed without merging**